adagrams/game.py

```diff
@@ -1,11 +1,139 @@
+import random
+#dictionary with the letters as keys and the allowed amount as values
+LETTERS_ALPHABET = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+#helper function to get a random letter from LETTERS_ALPHABET dictionary
+def random_selected_letter():
+    letters_alphabet = list(LETTERS_ALPHABET.keys())
+    #randint() is part of random library and random_index will return a random index between 0 and the higher limit in the length of letters_alphabet list
+    #every time we use this function it will retrieve a random index number
+    random_index = random.randint(0, len(letters_alphabet)-1)
+    #we'll use that random_index to retrieve the element (letter) that is in that position in the list
+    random_letter = letters_alphabet[random_index]
+
+    return random_letter
+
 def draw_letters():
-    pass
+    #hand variable will storage all the 10 random letters
+    hand = []
+    random_letter = ""
+
+    #while loop to loop through both lists letters_alphabet and LETTERS_MAX_ALLOWED to make
+    # sure we don't use more than the maximun amount allowed per letter
+    # while loop stops when we have 10 letters appended  
+    while len(hand) < 10:
+        random_letter= random_selected_letter()
+        if hand.count(random_letter) < LETTERS_ALPHABET[random_letter]:
```

Comment:
Using `count` does address the previous use of the the singe `cant_letter` variable. However, keep in mind that `count` must iterate over the hand each time through the draw loop to check for the number that letter appears (a hidden internal loop). Though we know the hand will never be more than 10 characters as written, in the general case this would take O(n) time where n is the size of the hand.

Rather than counting each time, or using a single counter, we could use a dictionary to track a count for each letter as you di with your `word_letter_dictionaries` below. This would allow us to track the count per letter, avoiding the single counter issue from before, and also avoiding needing an inner loop for counting.